### PR TITLE
🆕(site) Hero Animated Updates

### LINKED ISSUE #261: Module: Home Hero
Design: https://www.figma.com/file/xtBKZcRBo7H2Jrjs7DjNqE/Final-Designs?node-id=2%3A4714

### PR DESCRIPTION
**Pull request by nickalcantara**
Updates the hero animation and changes styling. Proactively adds EE animation data. Updates corresponding docs.

- Resolves #261

### CODE REVIEW

**Review by Snugug — CHANGES_REQUESTED** [review]
Thanks for your work on this! It all generally looks fine to me. I think I'm going to wait for VQA to get back to merge it, though, because I think there's too much block spacing on the animation when it drops underneath the text. I also don't see where `phosphor` gets swapped in.

**Comment by nickalcantara** [comment]
Thanks for the input - We've passed VQA and I've eliminated the row gap which should help a bit with the block spacing. The safe area for the animation is the only remaining space. I also added the phosphor integration.

File: site/js/components/hero-animated.js [hunk 1]
```diff
@@ -25,22 +25,34 @@ export class HeroAnimated {
   constructor(element) {
     // Create a set of constants to be reused throughout the class.
     const constants = {
-      wrapperSelector: '.hero-animated__animation',
-      animationItemClass: 'hero-animated__animation-item',
-      animationDelay: 2000,
-      animationName: 'Hero',
+      animation: {
+        delay: 2000,
```
Comment (Snugug) [inline site/js/components/hero-animated.js:29]: I know we had this delay in the original here, I'm not sure we need it now, though.

File: site/js/components/hero-animated.js [hunk 2]
```diff
@@ -53,25 +65,56 @@ export class HeroAnimated {
...
     this.init_();
   }
 
   /**
-   * Inits the animation and sets the event handlers.
+   * Check for user preferences and use the theme to apply the corresponding animation.
+   */
+  async useTheme() {
+    if (!window.matchMedia('(prefers-reduced-motion: reduce)').matches) {
+      const phosphor = document.body.dataset.theme === 'phosphor';
```
Comment (Snugug) [inline site/js/components/hero-animated.js:87]: The only place this is getting called (in the constructor in the event listener) passes in the theme name. Let's use that instead here

File: site/js/components/hero-animated.js [hunk 3]
```diff
@@ -53,25 +65,56 @@ export class HeroAnimated {
...
+      this.showMotionFallback();
+    }
+  }
+
+  /**
+   * Removes the class that hides the fallback asset and adds it to the background Image.
+   */
+  showMotionFallback() {
+    this.fallbackImage_.classList.remove(this.constants_.hideClass);
+    this.backgroundImage_.classList.add(this.constants_.hideClass);
```
Comment (Snugug) [inline site/js/components/hero-animated.js:108]: Doesn't `phosphor` also need to be hidden for motion fallback?

File: site/js/main.js [hunk 4]
```diff
@@ -57,9 +57,6 @@ window.addEventListener('DOMContentLoaded', async () => {
     offlineSearch.style.display = 'block';
   }
 
-  const { Theme } = await import('./components/theme');
-  new Theme();
```
Comment (Snugug) [inline site/js/main.js:61]: Why are you removing this from `main.js`? We need this to load phosphor on site load

File: site/js/components/theme.js [hunk 5]
```diff
@@ -29,6 +30,21 @@ export class Theme {
...
+    this.dispatch();
+  }
+
+  /**
+   * Dispatch a custom event with the current name.
+   * @return {void}
+   */
+  dispatch() {
+    this._elem.dispatchEvent(
+      new CustomEvent('themeApplied', {
```
Comment (Snugug) [inline site/js/components/theme.js:42]: Let's follow JS event naming conventions and make this `alllowercase` and name it as a change event

```suggestion
      new CustomEvent('themechange', {
```

File: site/js/components/hero-animated.js [hunk 6]
```diff
@@ -25,22 +25,34 @@ export class HeroAnimated {
...
+    this.fallbackImage_ = this.elem_.querySelector(this.constants_.selectors.fallback);
+    this.phosphorImage_ = this.elem_.querySelector(this.constants_.selectors.phosphor);
     this.animation_ = {};
 
     // Start the animation in case lottie loaded before the background image.
     if (!this.backgroundImage_.complete) {
       this.backgroundImage_.addEventListener('load', this.startIfReady_.bind(this));
     }
+
+    document.body.addEventListener('themeApplied', ({ detail }) => {
```
Comment (Snugug) [inline site/js/components/hero-animated.js:53]:
```suggestion
    document.body.addEventListener('themechange', ({ detail }) => {
```

**Review by Snugug — CHANGES_REQUESTED** [review]
Thanks for the updates here! A couple of comments w/r/t the phosphor stuff

File: site/js/components/hero-animated.js (hunk 3 above)
Comment (nickalcantara) [reply site/js/components/hero-animated.js:108]: The phosphor asset is hidden by default when it's not a descendant of an element with `data-theme="phosphor"` within `_hero-animated.scss`.

File: site/js/main.js (hunk 4 above)
Comment (nickalcantara) [reply site/js/main.js:61]: Turns out it was being called twice - `ee.js` calls a new Theme within its constructor, making this one redundant.

**Review by Snugug — APPROVED** [review]

**Pull request merged**